sim/dex.ts

```diff
@@ -570,6 +570,35 @@ dexes['base'] = new ModdedDex();
 dexes[BASE_MOD] = dexes['base'];
 
 export const Dex = dexes['base'];
+
+// Populate _toIDCache with data from the base mod.
+// It's representative enough for all other mods, so we don't need to dynamically grow the cache.
+{
+	const cache = Data._toIDCache;
+	const doName = (name?: string) => {
+		if (!name) return;
+		const id = toID(name);
+		const old = cache.get(name);
+		if (old === undefined) cache.set(name, id);
+		else if (old !== id) throw new Error("internal error with ID caching logic");
+	};
+	const sources: DataType[] = ['Abilities', 'Rulesets', 'Items', 'Moves', 'Natures', 'Pokedex', 'Conditions'];
+	for (const sourceName of sources) {
+		const source = Dex.data[sourceName] as any;
+		for (const k in source) {
+			doName(source[k].name);
+		}
+	}
+	for (const type of Dex.types.all()) {
+		doName(type.name);
+	}
+	const aliases = Dex.data.Aliases;
+	for (const k in aliases) {
+		doName(aliases[k]);
+	}
+	Object.freeze(cache);
```

Comment:
Freezing a ``Map`` has no (significant) effect, since its entries are stored in [[MapData]] internal slot.

Even something like ``cache.set = function () {throw new Error('owo'); }`` is pointless, because you can retrieve ``Map.prototype.set`` from elsewhere anyway.

Reply:
RIP.
Thanks for letting me know. Guess the underscore will have to suffice.

Reply:
Could just make it a non-exported variable in the ``toId`` scope, which makes it unaccessible to user-code.

Reply:
I'm confused about the loading order.

My understanding is that 
1) `toID` needs to exist before we can run the cache init logic.
2) having the cache captured by `toID`s closure in that manner would require defining `toID` where the cache is current populated (bottom of `./sim/dex.ts`).

Did I misunderstand you?

Reply:
Oh, you are right. It would need to be exported in a very annoying way in order to use a closure-based approach. And then somehow revoke the capability to update the cache from the exported object.

The simpler solution would be to ``delete Data._toIDCache``, I think? Oh, yea. It would be simpler if we were using CommonJS. But these are ES Imports (a clear downgrade from CommonJS, in my opinion.)

I tend to think that no freezing semantics are needed. But I guess it's up to you, if you want to go that extra mile.